src/renderer/dx/DxRenderer.cpp

```diff
@@ -1196,6 +1497,51 @@ enum class CursorPaintType
     return S_OK;
 }
 
+// Routine Description:
+// - Paint terminal effects.
+// Arguments:
+// Return Value:
+// - S_OK or relevant DirectX error.
+[[nodiscard]] HRESULT DxEngine::PaintTerminalEffects() noexcept
+{
+    assert(_framebufferCapture && "You should have initialized this");

```

Comment:
`assert` fires only in debug. We should have this be an `RETURN_HR_IF(E_STATE, _framebufferCapture)` or something.